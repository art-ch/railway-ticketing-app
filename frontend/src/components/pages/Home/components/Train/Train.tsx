'use client';

import React from 'react';

import { Train as TrainType } from 'railway-ticketing-app-sdk';
import Link from 'next/link';
import { getFormattedTime } from '@/utils/formatters';

export type TrainProps = { train: TrainType };

export const Train = ({ train }: TrainProps) => {
  const {
    trainId,
    name,
    departureStation,
    arrivalStation,
    departureTime,
    arrivalTime
  } = train;

  const formattedDepartureTime = getFormattedTime(new Date(departureTime));
  const formattedArrivalTime = getFormattedTime(new Date(arrivalTime));

  return (
    <Link
      href={`/${trainId}`}
      target="blank"
      className="text-white text-3xl p-3 font-dot-matrix font-bold grid grid-cols-[1fr_250px_250px_150px_150px] hover:bg-amber-300 hover:text-black hover:font-extrabold hover:cursor-pointer"
    >
      <div>{name}</div>
      <div>{departureStation}</div>
      <div>{arrivalStation}</div>
      <div>{formattedDepartureTime}</div>
      <div>{formattedArrivalTime}</div>
    </Link>
  );
};
