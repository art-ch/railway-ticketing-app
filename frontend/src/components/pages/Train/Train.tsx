'use client';

import React from 'react';

import { Seat } from './components/Seat/Seat';
import { useTrainPageContext } from '@/app/[trainId]/context';
import { WideScreenDemoNotice } from '@/components/WideScreenDemoNotice/WideScreenDemoNotice';
import { useBreakpoint } from '@/hooks/useBreakpoint/useBreakpoint';

export const Train = () => {
  const { trainData: trainInfo } = useTrainPageContext();

  const breakPoint = useBreakpoint();

  if (!breakPoint.xl) {
    return <WideScreenDemoNotice />;
  }

  if (!trainInfo) {
    return <div>Train not found</div>;
  }

  const { seats, trainId } = trainInfo;

  return (
    <main>
      <div className="h-[calc(100vh-66px)] flex items-center justify-center">
        <div className="border border-slate-300 rounded-xl p-7 grid grid-cols-5 gap-x-7 gap-y-10">
          {seats.map((seat) => (
            <Seat key={seat.seatNumber} trainId={trainId} seat={seat} />
          ))}
        </div>
      </div>
    </main>
  );
};
