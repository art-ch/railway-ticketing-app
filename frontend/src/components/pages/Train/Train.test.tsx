import { render } from '@testing-library/react';
import { Train } from './Train';
import {
  BreakpointState,
  useBreakpoint
} from '@/hooks/useBreakpoint/useBreakpoint';

jest.mock('@/app/[trainId]/context', () => ({
  useTrainPageContext: jest.fn().mockReturnValue({
    trainData: { trainId: '1', seats: [{ seatNumber: 1, isBooked: false }] }
  })
}));

jest.mock('@/hooks/useBreakpoint/useBreakpoint', () => ({
  useBreakpoint: jest.fn()
}));

const useBreakpointMock = useBreakpoint as jest.MockedFunction<
  typeof useBreakpoint
>;

describe('Train component', () => {
  it('should render correctly', () => {
    useBreakpointMock.mockReturnValue({ xl: true } as BreakpointState);

    const { container } = render(<Train />);

    expect(container).toMatchSnapshot();
  });

  it('should demo notice for small screen', () => {
    useBreakpointMock.mockReturnValue({ xl: false } as BreakpointState);

    const { container } = render(<Train />);

    expect(container).toMatchSnapshot();
  });
});
