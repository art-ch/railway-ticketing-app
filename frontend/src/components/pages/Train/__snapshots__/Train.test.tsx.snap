// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Train component should demo notice for small screen 1`] = `
<div>
  <div>
    This demo is intended to be shown on desktop
  </div>
</div>
`;

exports[`Train component should render correctly 1`] = `
<div>
  <main>
    <div
      class="h-[calc(100vh-66px)] flex items-center justify-center"
    >
      <div
        class="border border-slate-300 rounded-xl p-7 grid grid-cols-5 gap-x-7 gap-y-10"
      >
        <a
          class="border border-slate-300 rounded-sm p-1 text-center text-5xl cursor-pointer  text-white bg-emerald-800"
          href="/1/1"
        >
          1
        </a>
      </div>
    </div>
  </main>
</div>
`;
